src/main/java/org/apache/hadoop/fs/PrestoFileSystemCache.java

```diff
@@ -58,6 +60,11 @@ private synchronized FileSystem getInternal(URI uri, Configuration conf, Key key
             return fs;
         }
 
+        if (map.size() > CACHE_LEAK_PROTECTION_THRESHOLD) {
+            throw new IOException("Hadoop file system leak protection threshold has been reached. Threshold can be changed by setting " +
+                    "'-DhadoopFileSystemCacheLeakProtectionThreshold=<value>' jvm property. Current limit is: " + CACHE_LEAK_PROTECTION_THRESHOLD);
```

Comment:
I would rather leave the hint in the message of how to change this limit. What do you think?